nemo_skills/pipeline/verl/ppo.py

```diff
@@ -0,0 +1,380 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import os
+from dataclasses import dataclass
+from enum import Enum
+from typing import List, Optional
+
+import nemo_run as run
+import typer
+
+from nemo_skills.pipeline import add_task, check_if_mounted, get_cluster_config, run_exp
+from nemo_skills.pipeline.app import app, typer_unpacker
+from nemo_skills.pipeline.verl import verl_app
+from nemo_skills.pipeline.utils import get_free_port, get_ray_server_cmd, get_timeout
+from nemo_skills.utils import setup_logging
+
+LOG = logging.getLogger(__file__)
+
+
+@dataclass
+class PPOVerlTask:
+    model: str
+    reward_model: str
+    output_dir: str
+    prompt_data: str
+    eval_data: str
+    num_gpus: int
+    num_nodes: int
+    expname: str
+    disable_wandb: bool
+    wandb_project: str
+    timeout: str
+    extra_arguments: str = ""
+    logging_params: str = ""
+
+    def get_ray_launch_cmd(self):
+        cmd = "ray job submit --address='http://127.0.0.1:8265' -- "
+        return cmd
+
+    def format_train_args(self):
+        cmd = (
+            "   algorithm.adv_estimator=grpo "
+            "   data.train_batch_size=128 "
+            "   data.val_batch_size=512 "
+            "   data.max_prompt_length=1024 "
+            "   data.max_response_length=8192 "
+            f"   actor_rollout_ref.model.path={self.model} "
+            "   actor_rollout_ref.actor.optim.lr=1e-6 "
+            "   actor_rollout_ref.model.use_remove_padding=True "
+            "   actor_rollout_ref.actor.ppo_mini_batch_size=64 "
+            "   actor_rollout_ref.actor.use_dynamic_bsz=True "
+            "   actor_rollout_ref.actor.ppo_max_token_len_per_gpu=32768 "
+            "   actor_rollout_ref.actor.use_kl_loss=True "
```

Comment:
if default coef is 0, should we set this to False? Does it keep a reference model in memory if this is True?